### ISSUE TITLE
Support for using a closed file descriptor as a new file descriptor in dup2

### ISSUE THREAD

**Issue by OmSaran** [utsaslab/SplitFS#14]
When a closed file descriptor is attempted to be used as a new file descriptor in a `dup2` call, hub throws an error.

Here's a sample snippet to reproduce the issue.

```
#define FILE_PATH "/mnt/pmem_emul/test.txt"
#define FILE_PATH2 "/mnt/pmem_emul/test2.txt"

int main() {

  // Open the file
  int fd = open(FILE_PATH, O_CREAT | O_RDWR);
  int fd2 = open(FILE_PATH2, O_CREAT | O_RDWR);
  assert(fd >= 0);
  printf("fd = %d\n", fd);
  printf("fd2 = %d\n", fd2);

  // Write something onto it.
  char buf[] = "Writing to test.txt!\n";
  write(fd, buf, strlen(buf));

  // Close the file
  close(fd);

  // Try to dup2 using fd
 int ret =  dup2(fd2, fd);
 if(ret < 0) {
   perror("Failed to dup2!\n");
   exit(2);
 } else {
   // Write something onto test2 file using duped fd
   char buf2[] = "Writing something to test2.txt!\n";
   write(fd, buf2, strlen(buf2));
 }

 close(fd);
 close(fd2);
}
```

When a file is closed, the corresponding fileops lookup table in hub (_hub_fd_lookup[new_fd]) is [set to NULL](https://github.com/utsaslab/SplitFS/blob/master/splitfs/fileops_hub.c#L1104).
In dup2, there is a [check](https://github.com/utsaslab/SplitFS/blob/master/splitfs/fileops_hub.c#L1187) for whether it is NULL for the new file descriptor, and fails if this check is true.

Not sure why the fileops is set to NULL for the file descriptor being closed. 
Shouldn't it be set to the original state i.e use _hub_fileops (as done in `hub_check_resolve_fileops` [function](https://github.com/utsaslab/SplitFS/blob/master/splitfs/fileops_hub.c#L356))?

Am I missing something here?

**Comment by OmSaran** [comment]
#15 - Here is an attempt to address this

**Comment by OmSaran** [comment]
Solved by #16
